logger/include/logger_lib.hpp

```diff
@@ -0,0 +1,25 @@
+#ifndef LOGGER_LIB_HPP
+#define LOGGER_LIB_HPP
+
+#include <string>
+#include <memory>
+
+namespace logger
+{
+	/**
+	* @brief  Log level with their severity order , 0 =>high priority , 5=> least priority. 
+	*/
+	enum class LEVEL = { OFF = 0, FATAL = 1, ERROR = 2,WARN = 3,DEBUG = 4,TRACE = 5 };
+
+	/**
+	* @brief  Interface for the extracted log data like file name , function name etc , one may extend for more parameters such as number of argumnets __VA_ARGS__.
+	*/
+	struct IExtarctedLogCredentials
```

Comment:
what is the use of this structure and why it is added in this pull request

Reply:
this will be used to  retain the log credentials like date time , function name etc which are required by the recorder to log this information .